pkg/processor/batchprocessor.go

```diff
@@ -96,10 +97,11 @@ func (s *segmentsBatch) poll() {
 				for _, unprocessedSegment := range r.UnprocessedTraceSegments {
 					telemetry.T.SegmentRejected(1)
 					log.Errorf("Unprocessed segment: %v", unprocessedSegment)
-				}
-				log.Debug("Batch that contains unprocessed segments")
-				for i := 0; i < len(batch); i++ {
-					log.Debug(*batch[i])
+					for i := 0; i < len(batch); i++ {
```

Comment:
Ideally we'd wrap this with something like `isDebugEnabled` but I can't find anything like that in this log library. Actually, this log library is very old and seems of questionable quality (global lock on every log command, even disabled log levels? Blows my mind https://github.com/cihub/seelog/blob/f561c5e57575bb1e0a2167028b7339b3a8d16fb4/log.go#L208). So probably need to swap it out for zap

Reply:
agree, we'd better change a log lib before merging this code.